crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs

```diff
@@ -429,4 +430,19 @@ def with_lock():
         assert_eq!(get_complexity_number(&stmts), 2);
         Ok(())
     }
+
+    #[test]
+    fn match_case() -> Result<()> {
+        let source = r"
+def f():
+    match a:
+        case 30:
+            print('foo')
+        case _:
```

Comment:
I'm not so sure about this, the equivalent `else statement` has the following check: 

```rust
                   if clause.test.is_some() {
                        complexity += 1;
                    }
```

Reply:
Can you clarify what you're not sure about? Is it that the catch-all case (`_`) acts as an `else` block of an `if` statement?

Reply:
> Is it that the catch-all case (_) acts as an else block of an if statement?

Yep, complexity wise to me it's equivalent to the following `if else` statement.

```python
if x == 30:
    print("foo")
else:
    print("bar")
```

Reply:
Aren't they equivalent? We'd consider the above `if else` statement with complexity 2 and similarly for the following match statement:

```py
match subject:
	case foo: ...
	case _: ...
```

Reply:
The `if else` statement has a complexity of 1, the ` &clause.test.is_some()` evaluates to false for the `else`.

```sh
[crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs:413:9] &source = "\nif x == 3:\n    print(\"foo\")\nelse:\n    print(\"bar\")\n "
[crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs:79:21] &clause.test.is_some() = false
[crates/ruff_linter/src/rules/mccabe/rules/function_is_too_complex.rs:415:9] &get_complexity_number(&stmts) = 1
```

Reply:
Oh I see. Hmm, that's interesting. Yeah, I think we could do that i.e., check if the last `case` pattern is either a `_` or any variable as both act as a catch-all pattern.

Reply:
I think only `_` acts as a wild card pattern if my understanding of the [documentation](https://docs.python.org/3/reference/compound_stmts.html#wildcard-patterns) is correct. In any case, I can make a follow up PR to handle this. :)

Reply:
> I think only `_` acts as a wild card pattern if my understanding of the [documentation](https://docs.python.org/3/reference/compound_stmts.html#wildcard-patterns) is correct.

Yes, although a name pattern would also _act_ in a similar way except that it also binds the value to the variable. So, in the following example `foo` will be same as `subject`. It's like doing `foo = subject`.

```py
match subject:
	case 2: ...
	case foo: ...
```

You can even see that in CPython where it raises a syntax error similar to `_` if the name pattern is not the last pattern:

```py
match 1:
    case x:
        print(x)
    case 2:
        ...
```

Running `python test.py`:

```console
$ python3.13 parser/_.py
  File "/Users/dhruv/playground/ruff/parser/_.py", line 2
    case x:
         ^
SyntaxError: name capture 'x' makes remaining patterns unreachable
```

> In any case, I can make a follow up PR to handle this. :)

Sounds good.